web/ui/react-app/src/Checkbox.tsx

```diff
@@ -0,0 +1,23 @@
+import React, { FC, useState, HTMLProps, memo } from 'react';
+import { FontAwesomeIcon, FontAwesomeIconProps } from '@fortawesome/react-fontawesome';
+
+interface CheckboxProps extends Omit<HTMLProps<HTMLSpanElement>, 'size' | 'onClick'> {
+  onToggle: (selected: boolean) => void
+}
+
+const Checkbox: FC<CheckboxProps & Pick<FontAwesomeIconProps, 'size'>> = props => {
```

Comment:
I learned new TypeScript + React things in this file, thanks :laughing:

Reply:
Yes, generally this will go away since I'm going to replace it with reactstrap Input